src/pip/_internal/vcs/subversion.py

```diff
@@ -37,7 +54,57 @@ def should_add_vcs_url_prefix(cls, remote_url):
     def get_base_rev_args(rev):
         return ['-r', rev]
 
+    def vcs_global_options(self):
+        # type: () -> List[str]
+        """Options to be used on calls to Subversion that contact the server.
+
+        These options are applicable for the following ``svn`` subcommands used
+        in this class.
+
+            - checkout
+            - export
+            - info
+            - switch
+            - update
+
+        :return: A list of command line arguments to pass to ``svn``.
+        """
+        options = []  # type: List[str]
+        if not self.use_interactive:
+            # --non-interactive switch is available since Subversion 0.14.4.
+            options.append('--non-interactive')
+        else:
+            svn_version = self.get_vcs_version()
+            # Cannot safely add new options if SVN version is unknown.
+            # --force-interactive was added in Subversion 1.8.
+            # RHEL/CentOS 7 ships with Subversion 1.7 and is supported until
+            # 2024.
+            # By default, Subversion 1.8 runs non-interactive if stdin
+            # is not a TTY. Since that is how pip invokes Subversion,
+            # it must pass --force-interactive to ensure the user can be
+            # prompted for a password, if required.
+            if svn_version is not None and svn_version >= (1, 8):
+                options.append('--force-interactive')
+        return options
+
     def get_vcs_version(self):
+        # type: () -> Optional[Tuple[int, ...]]
+        if self._vcs_version is not None:
+            # Use cached version, if available.
+            return self._vcs_version
+        elif self._call_vcs_version_failed:
+            # If parsing the version failed previously, do not attempt
+            # to parse it again.
+            return None
+        else:
+            vcs_version = self.call_vcs_version()
+            if vcs_version is not None:
+                self._vcs_version = vcs_version
+            else:
+                self._call_vcs_version_failed = True
+            return vcs_version
+
+    def call_vcs_version(self):
```

Comment:
In terms of ordering methods, I think it helps to follow the rule to define methods above where they are used. That way you can read classes top to bottom. So `call_vcs_version()` then `get_vcs_version()` then `get_vcs_global_options()`.